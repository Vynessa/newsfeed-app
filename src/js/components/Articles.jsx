import React from 'react';
import { browserHistory } from 'react-router';
import NewsActions from '../actions/newsActions';
import ArticlesStore from '../stores/articlesStore';
import ArticlesList from './ArticlesList.jsx';

/**
 * @description Articles component
 * @class
 */
class Articles extends React.Component {
  /**
   * Initialize State
   * @param {object} props
   */
  constructor(props) {
    super(props);
    this.state = {
      articles: ArticlesStore.getAll(),
      sortBy: '',
      user: null
    };
    this.sourceKey = ArticlesStore.getSourceKey() || localStorage.getItem('sourceKey');
    this.onChange = this.onChange.bind(this);
    this.updateSortBy = this.updateSortBy.bind(this);
    this.handleSort = this.handleSort.bind(this);
  }

  /**
   * @description
   * @returns {void}
   * @memberof Sources
   */
  componentWillMount() {
    const user = JSON.parse(localStorage.getItem('user'));
    this.setState({
      user
    });
  }
  /**
   * @description Mount Component on every change
   * @method
   * @returns {void}
   */
  componentDidMount() {
    // NewsActions.allArticles(this.sourceKey, sortBy);
    ArticlesStore.on('change', this.onChange);
  }
  /**
   * @description Unmount Component
   * @method
   * @returns {void}
   */
  componentWillUnmount() {
    ArticlesStore.removeListener('change', this.onChange);
  }
  /**
   * @description Change state on user(s) request
   * @method
   * @returns {object} articles
   */
  onChange() {
    this.setState({
      articles: ArticlesStore.getAll(),
    });
  }
  /**
   *@description Sort function to filter headlines
   * @method
   * @param {string} sourceKey
   * @param {string} sortBy
   * @returns {object} articles
   */
  handleSort(sourceKey, sortBy) {
    sourceKey = localStorage.getItem('sourceKey');
    this.setState({
      sourceKey
    });
    return NewsActions.allArticles(sourceKey, sortBy);
  }
  /**
   * @description Event handler function to update sort state
   * @method
   * @param {object} event
   * @returns {string} sortBy
   */
  updateSortBy(event) {
    const val = event.target.value;
    this.setState({
      sortBy: val
    });
  }
  /**
   * @description render child component - ArticlesList
   * @method
   * @returns {JSX.Element} ArticlesList
   */
  render() {
    return (
      <ArticlesList
        articles={this.state.articles}
        handleSort={this.handleSort}
        sortBy = {this.state.sortBy} />
    );
  }
}

export default Articles;
