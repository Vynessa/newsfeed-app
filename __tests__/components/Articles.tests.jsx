import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Articles from '../../src/js/components/Articles.jsx';
import mockedArticles from '../../__mocks__/mockArticles.json';

const sortBys = ['top', 'latest'];
const sourceKey = 'al-jazeera-english';
localStorage.setItem('sourceKey', sourceKey);
localStorage.setItem('sortBys', JSON.stringify(sortBys));

describe('Articles Component Test Suite', () => {
  test('should match the Articles snapshot', () => {
    const component = mount(<Articles />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });

  // test('', () => {
  //   const title = shallow(<Articles />);
  //   expect(title.find('h1').text()).toEqual('Headlines from Al Jazeera English');
  // });
  // test('should contain all articles based on the value in the state', () => {
  //   // const sourceId = localStorage.getItem('sourceKey');
  //   // const sortOpt = JSON.parse(localStorage.sortBys);
  //   const component = mount(<Articles articles={mockedArticles}/>);
  //   component.setState({
  //     mockedArticles
  //   });
  //   component.find('div').simulate('card');
  //   expect(toJson(component)).toMatchSnapshot();
  // });
});
