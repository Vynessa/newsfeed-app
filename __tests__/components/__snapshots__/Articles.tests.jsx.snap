// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles Component Test Suite should match the Articles snapshot 1`] = `
<Articles>
  <ArticlesList
    articles={Array []}
    handleSort={[Function]}
    sortBy=""
    source=""
    sourceKey="al-jazeera-english"
  >
    <div>
      <div
        className="clearfix"
      />
      <Breadcrumb
        className="fixed"
        cols={12}
      >
        <nav
          className="row"
        >
          <div
            className="nav-wrapper"
          >
            <Col
              s={12}
            >
              <div
                className="col s12"
              >
                <MenuItem
                  className="breadcrumb"
                  href="sources"
                  id="sources"
                >
                  <a
                    className="breadcrumb"
                    href="sources"
                    id="sources"
                  >
                    Sources
                  </a>
                </MenuItem>
                <MenuItem
                  className="breadcrumb"
                  id="articles"
                >
                  <a
                    className="breadcrumb"
                    id="articles"
                  >
                    Articles
                  </a>
                </MenuItem>
              </div>
            </Col>
          </div>
        </nav>
      </Breadcrumb>
      <h1
        className="center-align"
        id="heading-text"
      >
        Headlines from 
        Al Jazeera English
      </h1>
      <Row
        className="center-align"
      >
        <div
          className="row center-align"
        >
          <Input
            label="Sort Articles By:"
            m={6}
            onChange={[Function]}
            s={12}
            type="select"
          >
            <div
              className="col input-field s12 m6"
            >
              <label
                className="active"
                htmlFor="input_0"
              >
                Sort Articles By:
              </label>
              <select
                className=""
                id="input_0"
                onChange={[Function]}
              >
                <option
                  value="top"
                >
                  Top
                </option>
                <option
                  value="latest"
                >
                  Latest
                </option>
              </select>
            </div>
          </Input>
        </div>
      </Row>
      <div
        className="row"
      >
        <div
          className="center-align"
        >
          <Preloader>
            <div
              className="preloader-wrapper big active"
            >
              <div
                className="spinner-layer spinner-brown-only"
              >
                <div
                  className="circle-clipper left"
                >
                  <div
                    className="circle"
                  />
                </div>
                <div
                  className="gap-patch"
                >
                  <div
                    className="circle"
                  />
                </div>
                <div
                  className="circle-clipper right"
                >
                  <div
                    className="circle"
                  />
                </div>
              </div>
            </div>
          </Preloader>
        </div>
      </div>
    </div>
  </ArticlesList>
</Articles>
`;
